rtgui/colortoning.cc

```diff
@@ -1195,3 +1372,195 @@ bool ColorToning::resetPressed(GdkEventButton* event)
+    labRegionPopulateList();
+    labRegionShow(labRegionSelected);
+
+    if (listener) {
+        listener->panelChanged(EvLabRegionList, M("HISTORY_CHANGED"));
+    }
+}
+
+
+void ColorToning::labRegionRemovePressed()
+{
+    if (labRegionList->size() > 1) {
+        labRegionData.erase(labRegionData.begin() + labRegionSelected);
+        labRegionSelected = LIM(labRegionSelected-1, 0, int(labRegionData.size()-1));
+        labRegionPopulateList();
+        labRegionShow(labRegionSelected);
+
+        if (listener) {
+            listener->panelChanged(EvLabRegionList, M("HISTORY_CHANGED"));
+        }
+    }
+}
+
+
+void ColorToning::labRegionUpPressed()
+{
+    if (labRegionSelected > 0) {
+        auto r = labRegionData[labRegionSelected];
+        labRegionData.erase(labRegionData.begin() + labRegionSelected);
+        --labRegionSelected;
+        labRegionData.insert(labRegionData.begin() + labRegionSelected, r);
+        labRegionPopulateList();
+
+        if (listener) {
+            listener->panelChanged(EvLabRegionList, M("HISTORY_CHANGED"));
+        }
+    }
+}
+
+
+void ColorToning::labRegionDownPressed()
+{
+    if (labRegionSelected < int(labRegionData.size()-1)) {
+        auto r = labRegionData[labRegionSelected];
+        labRegionData.erase(labRegionData.begin() + labRegionSelected);
+        ++labRegionSelected;
+        labRegionData.insert(labRegionData.begin() + labRegionSelected, r);
+        labRegionPopulateList();
+
+        if (listener) {
+            listener->panelChanged(EvLabRegionList, M("HISTORY_CHANGED"));
+        }
+    }
+}
+
+
+void ColorToning::labRegionShowMaskChanged()
+{
+    if (listener) {
+        listener->panelChanged(EvLabRegionShowMask, labRegionShowMask->get_active() ? M("GENERAL_ENABLED") : M("GENERAL_DISABLED"));
+    }
+}
+
+
+namespace {
+
+bool hasMask(const std::vector<double> &dflt, const std::vector<double> &mask)
+{
+    if (mask.empty() || mask[0] == FCT_Linear || mask == dflt) {
+        return false;
+    } else {
+        return true;
+    }
+}
+
+} // namespace
+
+
+void ColorToning::labRegionPopulateList()
+{
+    ConnectionBlocker b(labRegionSelectionConn);
+    labRegionList->clear_items();
+    rtengine::ColorToningParams::LabCorrectionRegion dflt;
+    
+    for (size_t i = 0; i < labRegionData.size(); ++i) {
+        auto &r = labRegionData[i];
+        auto j = labRegionList->append(std::to_string(i+1));
+        labRegionList->set_text(j, 1, Glib::ustring::compose("a=%1 b=%2 s=%3 l=%4", int(r.a), int(r.b), r.saturation, r.lightness));
+        labRegionList->set_text(
+            j, 2, Glib::ustring::compose(
+                "%1%2%3",
+                hasMask(dflt.hueMask, r.hueMask) ? "H" : "",
+                hasMask(dflt.chromaticityMask, r.chromaticityMask) ? "C" : "",
+                hasMask(dflt.lightnessMask, r.lightnessMask) ? "L" : ""));
+    }
+}
+
+
+void ColorToning::labRegionShow(int idx, bool list_only)
+{
+    bool disable = listener;
```

Comment:
Could be `const`. (Same applies to many other instances I've read above, but this here is most obvious.)